src/users/users.service.ts

```diff
@@ -12,11 +13,25 @@ export class UsersService {
   ) {}
 
   async create(createUserDto: CreateUserDto): Promise<User & Document> {
-    const createdUser = new this.userModel(createUserDto);
-    return createdUser.save();
+    try {
+      const hashedPassword = await bcrypt.hash(createUserDto.password, 2);
+      const newUser = {
+        email: createUserDto.email,
+        username: createUserDto.username,
+        hash: hashedPassword,
+      };
+      const createdUser = new this.userModel(newUser);
+      return createdUser.save();
+    } catch (e) {
+      throw new Error(e);
+    }
   }
 
-  async findOne(username: CreateUserDto['username']): Promise<User> {
-    return this.userModel.findOne({ username }).lean();
+  async findOne(user: Partial<CreateUserDto>): Promise<User | null> {
+    try {
+      return this.userModel.findOne({ username: user.username }).lean();
```

Comment:
should this be awaited?

Reply:
All async functions return automatically a Promise, as I don't need to process the value before I return I can directly forward the Promise returned by `findOne`, whoever wants to look inside will have to `await` it.

Reply:
I'll remove the `try/catch` as this Promise cannot be rejected